main/src/main/scala/sbt/Project.scala

```diff
@@ -157,6 +165,288 @@ private[sbt] object CompositeProject {
 
 }
 
+/**
+ * A project matrix is an implementation of a composite project
+ * that represents cross building across some axis (such as platform)
+ * and Scala version.
+ *
+ *  {{{
+ *  lazy val core = (projectMatrix in file("core"))
+ *    .scalaVersions("2.12.6", "2.11.12")
+ *    .settings(
+ *      name := "core"
+ *    )
+ *    .jvmPlatform()
+ *  }}}
+ */
+sealed trait ProjectMatrix extends CompositeProject {
+  def scalaVersions(sv: String*): ProjectMatrix
+
+  def id: String
+
+  /** The base directory for the project matrix.*/
+  def base: File
+
+  def withId(id: String): ProjectMatrix
+
+  /** Sets the base directory for this project matrix.*/
+  def in(dir: File): ProjectMatrix
+
+  /** Adds new configurations directly to this project.  To override an existing configuration, use `overrideConfigs`. */
+  def configs(cs: Configuration*): ProjectMatrix
+
+  /** Adds classpath dependencies on internal or external projects. */
+  def dependsOn(deps: MatrixClasspathDep[ProjectMatrixReference]*): ProjectMatrix
+
+  /**
+   * Adds projects to be aggregated.  When a user requests a task to run on this project from the command line,
+   * the task will also be run in aggregated projects.
+   */
+  def aggregate(refs: ProjectMatrixReference*): ProjectMatrix
+
+  /** Appends settings to the current settings sequence for this project. */
+  def settings(ss: Def.SettingsDefinition*): ProjectMatrix
+
+  /**
+   * Sets the [[AutoPlugin]]s of this project.
+   * A [[AutoPlugin]] is a common label that is used by plugins to determine what settings, if any, to enable on a project.
+   */
+  def enablePlugins(ns: Plugins*): ProjectMatrix
+
+  /** Disable the given plugins on this project. */
+  def disablePlugins(ps: AutoPlugin*): ProjectMatrix
+
+  def custom(
+      idPostfix: String,
```

Comment:
I think the word is suffix, not postfix? I haven't checked a dictionary but postfix is usually used for postfix operators. Appending to a string or word is usually called suffix.

Reply:
Yes. You're totally right.